### DISCUSSION TITLE
Configuration Variables

### DISCUSSION THREAD

**Discussion by andrewrynhard** [siderolabs/talos#3234] [Ideas]
Creating generic machine configuration files is difficult. As a cluster administrator I want to create generic configuration files that can be applied per node type instead of maintaining unique files.

Here is a list of possible solutions:

- Booleans that enable/disable high level functionality
- Well-known variables that will be parsed
- Client-side tooling (e.g. kustomize)

**Comment by DWSR** [comment]
> Creating genetic machine configuration files is difficult. As a cluster administrator I want to create generic configuration files that can be applied per node type instead of maintaining unique files.
> 
> Here is a list of possible solutions:
> 
> - Booleans that enable/disable high level functionality
> - Well-known variables that will be parsed
> - Client-side tooling (e.g. kustomize)

Something like how Consul and Vault read their config might be interesting, especially around go-sockaddr: https://www.consul.io/docs/agent/options
